### PR TITLE
ACC-1195 remove n-internal-tool, html engine fixes needed

### PR DESCRIPTION
**Pull request by asugar13**
Remove n-internal-tool and use n-express instead.

https://financialtimes.atlassian.net/jira/software/c/projects/ACC/boards/1461?modal=detail&selectedIssue=ACC-1195

### CODE REVIEW

File: demos/demo.html [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<link rel="stylesheet" href="/public/main.css">
```
Comment (asugar13) [inline demos/demo.html:1]: **Question**: I didn't like needing to do this. I was expecting a manifest.json file in the `public` folder. I think I would need to do this on every single html file.

File: demos/app.js [hunk 2]
```diff
@@ -19,9 +24,50 @@ const app = module.exports = express({
...
 	layoutsDir: 'demos',
 	partialsDirectory: process.cwd(),
-	directory: process.cwd()
+	directory: process.cwd(),
+});
+
+const templatePartialDirectory = path.join(__dirname, '../templates/partials');
+fs.readdirSync(templatePartialDirectory).forEach(filename => {
+	handlebars.registerPartial(
+		`templates/${filename.substr(0, filename.lastIndexOf('.'))}`,
```
Comment (magsallen) [inline demos/app.js:33]: [Copy/Pasta] It looks like this should switch places with line 42 to become `templates/partials/...`

File: demos/app.js [hunk 3]
```diff
@@ -19,9 +24,50 @@ const app = module.exports = express({
...
+		fs.readFileSync(path.join(templatePartialDirectory, filename), 'utf8')
+	);
+});
+
+const templateDirectory = path.join(__dirname, '../templates');
+fs.readdirSync(templateDirectory).forEach(filename => {
+	if (filename.includes('html')){
+		handlebars.registerPartial(
+			`templates/partials/${filename.substr(0, filename.lastIndexOf('.'))}`,
```
Comment (magsallen) [inline demos/app.js:42]: [Copy/Pasta] It looks like this should switch places with line 33 to become `templates/...`

**Review by magsallen — COMMENTED** [review]
This is impressive! But I think there's some functionality in Page Kit Handlebars that could simplify things. Have you considered using the `loadPartials` and `partialPaths` options? https://github.com/Financial-Times/dotcom-page-kit/tree/main/packages/dotcom-server-handlebars#loadpartials. My hunch is that this would take away some of the work of decoding the template paths manually. 

Ultimately, these changes only affect the demo app, so I'm happy enough and not too worried about it being perfect. Definitely time box any further investigations and revert to this solution if it's too clunky

File: demos/demo.html [hunk 4]
```diff
@@ -1,19 +1,28 @@
...
-	<div class="o-grid-row">
-		<div data-o-grid-colspan="12 S6 M4 L3">
-			{{> n-newsletter-signup/templates/newsletter flags=@root.flags}}
-		</div>
-	</div>
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<title>demo</title>
+		<link rel="stylesheet" href="/public/main.css">
```
Comment (asugar13) [inline demos/demo.html:5]: @magsallen I would like to chat about this. There's no manifest.json file being used in this repo. and I think I will need to do this on every single html file for the CSS to work.

File: demos/app.js (hunk 2 above)
Comment (asugar13) [reply demos/app.js:33]: good spot

File: demos/app.js (hunk 3 above)
Comment (asugar13) [reply demos/app.js:42]: good spot, updating PR

**Pull request merged**